### PR TITLE
Hoist the getModel function and allow it to be passed in

### PR DESCRIPTION
**Pull request by micahr**
I hoisted the getModel function out of the `forEachParallel` call so that a new function can be passed in if desired.

### CODE REVIEW

**Comment by micahr** [comment]
@DonutEspresso mind taking a quick look? I think I might break out more of the functionality in `getModel` so I can share it without having to copy/pasta into my app. But this gets the ball rolling.

File: lib/handlers/buildModels.js [hunk 1]
```diff
@@ -37,93 +152,13 @@ function buildModelsWrapper(modelBucket) {
             log = logHelpers.getDefault();
         }
 
+        partialModel = _.partial(modelFetcher, req, res);
```
Comment (DonutEspresso) [inline lib/handlers/buildModels.js:156]: No bind necessary? Amaze, so functional.

File: lib/handlers/buildModels.js [hunk 2]
```diff
@@ -3,12 +3,117 @@
...
+/**
+ * Internal method for fetching any models that are resolved from
+ * the model bucket.
+ *
+ * @private
+ * @param {Object} req request object
+ * @param {Object} res response object
+ * @param {Object} model model object
+ * @param {Function} callback callback when fetch is done
+ * @returns {undefined}
```
Comment (DonutEspresso) [inline lib/handlers/buildModels.js:23]: Nitpick, @function getModel.

Maybe this repo doesn't have JSDoc linting enforced.

**Comment by DonutEspresso** [comment]
Looks reasonable to me, git 'er rolling.

File: lib/handlers/buildModels.js (hunk 1 above)
Comment (micahr) [inline lib/handlers/buildModels.js:156]: I'm not binding the `this`, just partially applying the first two parameters. This really is much cleaner.

File: lib/handlers/buildModels.js (hunk 2 above)
Comment (micahr) [inline lib/handlers/buildModels.js:23]: fixed

**Comment by micahr** [comment]
Fixes #15

**Pull request merged**